core/src/test/scala/org/apache/spark/SparkThrowableSuite.scala

```diff
@@ -73,7 +74,17 @@ class SparkThrowableSuite extends SparkFunSuite {
 
   test("SQLSTATE invariants") {
     val sqlStates = errorClassToInfoMap.values.toSeq.flatMap(_.sqlState)
-    checkCondition(sqlStates, s => s.length == 5)
+    val errorClassReadMe = Utils.getSparkClassLoader.getResource("error/README.md")
+    val errorClassReadMeContents = IOUtils.toString(errorClassReadMe.openStream())
+    val sqlStateTableRegex =
+      "(?s)\\[//]: # \\(SQLSTATE table start\\)(.+)\\[//]: # \\(SQLSTATE table end\\)".r
+    val sqlTable = sqlStateTableRegex.findFirstIn(errorClassReadMeContents).get
+    val sqlTableRows = sqlTable.split("\n").filter(_.startsWith("|")).drop(2)
+    val validSqlStates = sqlTableRows.map(_.slice(1, 6)).toSet
+    // Sanity check
+    assert(Set("07000", "42000", "HZ000").subsetOf(validSqlStates), validSqlStates)
```

Comment:
Sorry maybe I missed something. But is there an easy to upload all existing error codes in Spark, and validate it?

Reply:
We validate the SQLSTATEs in the JSON below. This just checks that we extracted SQLSTATEs from the table.